File: G_Projects/I_Vending_Machine/PaymentStrategy/ConcreteStrategy/CoinPaymentStrategy.java
package G_Projects.I_Vending_Machine.PaymentStrategy.ConcreteStrategy;

import java.util.List;

import G_Projects.I_Vending_Machine.CommonEnums.Coin;
import G_Projects.I_Vending_Machine.PaymentStrategy.PaymentStrategy;

public class CoinPaymentStrategy implements PaymentStrategy {
    private List<Coin> insertedCoins;

    public CoinPaymentStrategy(List<Coin> insertedCoins){
        this.insertedCoins = insertedCoins;
    }
    @Override
    public boolean processPayment(int amount){
        int total = 0;
        for(Coin coin : insertedCoins){
            total += coin.getValue();
        }
        return total >= amount; // Check if the total value of inserted coins is greater then what needs to be paid
    }
}

File: Python/A_Project/E_Vending_Machine/PaymentStrategy/ConcreteStrategy/coin_payment_strategy.py
from PaymentStrategy.payment_strategy import PaymentStrategy


class CoinPaymentStrategy(PaymentStrategy):
    def __init__(self, inserted_coins):
        self.inserted_coins = inserted_coins

    def process_payment(self, amount: int) -> bool:
        total = sum(coin.get_value() for coin in self.inserted_coins)
        return total >= amount
